dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -823,17 +824,27 @@ static std::shared_ptr<InterpreterSelectWithUnionQuery> interpretSubquery(
         select_query->select_expression_list = select_expression_list;
         select_query->children.emplace_back(select_query->select_expression_list);
 
+        NamesAndTypesList columns;
+
         /// get columns list for target table
-        auto database_table = getDatabaseAndTableNameFromIdentifier(*table);
-        const auto & storage = context.getTable(database_table.first, database_table.second);
-        const auto & columns = storage->getColumns().ordinary;
-        select_expression_list->children.reserve(columns.size());
+        if (function)
+        {
+            const auto & storage = const_cast<Context *>(&context)->executeTableFunction(table_expression);
```

Comment:
That looks suspicious.
We need to either make function `executeTableFunction` const or figure out why it is non const and maybe change `context` in Expression analyzer to non-const and proove that nobody rely on it being const.

Reply:
👍 Indeed, the correct way to call it is `context->getQueryContext()->executeTableFunction`. 
Because `executeTableFunction` method always cache the current execution results. direct call to `context->executeTableFunction` could be a  memory leak.

Reply:
reference : https://github.com/yandex/ClickHouse/blob/master/dbms/src/Interpreters/InterpreterSelectQuery.cpp#L165